### PR TITLE
Refactor mixer

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
Doesn't build for me, but I did add the files to the project

**Comment by Gymnasiast** [comment]
```
src\audio\FileAudioSource.cpp(53): warning C4244: 'return': conversion from 'uint64' to 'std::size_t', possible loss of data [C:\projects\openrct2-ject9\openrct2.vcxproj]
src\audio\FileAudioSource.cpp(67): warning C4244: 'argument': conversion from 'uint64' to 'std::size_t', possible loss of data [C:\projects\openrct2-ject9\openrct2.vcxproj]
```

Seems to affect 32-bit Windows builds only.

File: src/audio/AudioChannel.cpp [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    {
+        return _rate;
+    }
+
+    void SetRate(double rate) override
+    {
+        _rate = Math::Max(0.001, rate);
+    }
+
+    unsigned long GetOffset() const override
```
Comment (marijnvdwerf) [inline src/audio/AudioChannel.cpp]: Shouldn't these be `uint64`?

**Comment by janisozaur** [comment]
@IntelOrca you should include tests for refactored code.

**Comment by IntelOrca** [comment]
@janisozaur I don't have time

File: src/audio/mixer.cpp [hunk 2]
```diff
@@ -14,939 +14,658 @@
...
+        _nullSource = AudioSource::CreateNull();
+    }
+
+    ~AudioMixer()
+    {
+        Close();
+        delete _nullSource;
+    }
+
+    void Init(const char* device) override
```
Comment (marijnvdwerf) [inline src/audio/mixer.cpp]: `utf8`/`uint8`/`sint8`?

File: src/audio/mixer.h [hunk 3]
```diff
@@ -17,177 +17,52 @@
...
+    virtual void Init(const char * device) abstract;
+    virtual void Close() abstract;
+    virtual void Lock() abstract;
+    virtual void Unlock() abstract;
+    virtual IAudioChannel * Play(IAudioSource * source, int loop, bool deleteondone, bool deletesourceondone) abstract;
+    virtual void Stop(IAudioChannel * channel) abstract;
+    virtual bool LoadMusic(size_t pathid) abstract;
+    virtual void SetVolume(float volume) abstract;
+
+    virtual IAudioSource * GetSoundSource(int id) abstract;
```
Comment (marijnvdwerf) [inline src/audio/mixer.h]: `uint32`/`sint32`?

File: src/audio/MemoryAudioSource.cpp [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+    bool Convert(const AudioFormat * format)
+    {
+        if (*format != _format)
+        {
+            SDL_AudioCVT cvt;
+            if (SDL_BuildAudioCVT(&cvt, _format.format, _format.channels, _format.freq, format->format, format->channels, format->freq) >= 0)
+            {
+                cvt.len = (int)_length;
+                cvt.buf = (Uint8*)new uint8[cvt.len * cvt.len_mult];
```
Comment (marijnvdwerf) [inline src/audio/MemoryAudioSource.cpp]: `Uint8 *` or `Uint8*`? Asterisk spacing seems to be inconsistent

**Comment by marijnvdwerf** [comment]
@janisozaur Doesn't clang-format run automatically for PR's now? I'd expect it to complain about different styles of `uint8 * name`

**Comment by janisozaur** [comment]
@marijnvdwerf no, it's to be used manually. There are inconsistencies even in our current code.

**Pull request merged**